src/main/flight/gyroanalyse.c

```diff
@@ -269,62 +311,64 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state,
         }
         case STEP_CALC_FREQUENCIES:
         {
-            bool fftIncreased = false;
-            float dataMax = 0;
-            uint8_t binStart = 0;
-            uint8_t binMax = 0;
-            //for bins after initial decline, identify start bin and max bin 
+            // identify max bin and max/min heights 
+            float dataMax = 0.0f;
+            float dataMin = 1.0f;
+            uint8_t binMax = 0; 
             for (int i = fftStartBin; i < FFT_BIN_COUNT; i++) {
-                if (fftIncreased || (state->fftData[i] > state->fftData[i - 1])) {
-                    if (!fftIncreased) {
-                        binStart = i; // first up-step bin
-                        fftIncreased = true;
-                    }
+                if (state->fftData[i] > state->fftData[i - 1]) { // bin height increased
                     if (state->fftData[i] > dataMax) {
                         dataMax = state->fftData[i];
+                        dataMin = state->fftData[i];
                         binMax = i;  // tallest bin
                     }
                 }
             }
-            // accumulate fftSum and fftWeightedSum from peak bin, and shoulder bins either side of peak
-            float cubedData = state->fftData[binMax] * state->fftData[binMax] * state->fftData[binMax];
-            float fftSum = cubedData;
-            float fftWeightedSum = cubedData * (binMax + 1);
-            // accumulate upper shoulder
-            for (int i = binMax; i < FFT_BIN_COUNT - 1; i++) {
-                if (state->fftData[i] > state->fftData[i + 1]) {
-                    cubedData = state->fftData[i] * state->fftData[i] * state->fftData[i];
-                    fftSum += cubedData;
-                    fftWeightedSum += cubedData * (i + 1);
-                } else {
-                break;
+            if (binMax == 0) { // no bin increase, hold prev max bin, dataMin = 1 dataMax = 0, ie move slow
+                binMax = lrintf(state->centerFreq[state->updateAxis] / fftResolution);
+            } else { // there was a max, look backwards for min, don't compare to bin 0
+                for (int i = binMax; i > 1; i--) {
```

Comment:
i > 1 will skip not only bin 0, but also bin 1.
Is this intended?

Reply:
As I intended it, if i > 1, the lowest possible value for i would be 2, and i-1 is 1.
Then we can compare bin 2 to bin 1, but not any lower.
That is intended to stop any comparison between bin 1 and bin 0, and I think it needs to be like this, since bin 0 does not carry any AC (frequency related) information.